#163604603 GET /parties endpoint

frankly034 opened the pull request:
#### What does this PR do?
Have api endpoint for GET /parties

#### Description of Task to be completed?
* Create party model function to fetch all parties from the resource
* Create party controller function to manage fetch all party requests
* Create party route to handle GET /parties request
* Create party create test

#### How should this be manually tested?
After cloning the repo, run 'npm run test'

#### Any background context you want to provide?
* Http test were done using supertest and chai expect library

#### What are the relevant pivotal tracker stories?
#163604603

frankly034 committed:
chore(officeModel): add findAll()

- add function to fetch all dummy data
[Delivers #163604603]

frankly034 committed:
chore(partyController): add listParty()

- add controller function to manage fetch all parties operation
[Delivers #163604603]

frankly034 committed:
feat(route): add GET /parties

- handle fetching all parties from the dummy data
[Delivers #163604603]

frankly034 committed:
feat(controller): add listparties() and test

- add listparties () to manage fetching all parties
- implemented test for the endpoint
[Delivers #163604603]

Lumexralph commented on server/models/partyModel.js:
```diff
@@ -14,6 +14,10 @@ class PartyModel {
     parties.push(newParty);
     return newParty;
   }
```
@frankly034 Nice work! since this file is holding a `class` I will suggest you change the name of the file from `camelCase` to `pascal`

frankly034 replied:
Ok, I'll modify all of them. Thanks

frankly034 replied:
Thanks for the observation. Modified, committed and pushed.

frankly034 committed:
bug(file names): change file naming convention

- changed all the file names with classes to pascal case
- adjusted all the references for those files
[Delivers #163604603]

frankly034 commented:
Merging based on your message in the channel. Thanks for everything.
